<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="../../../css/estrutura.css">
    <link rel="stylesheet" href="../../../css/style.css">
    <title>Tratando e lançando erros (try, cath, throw)</title>
    <script>
        var active = 4
        var voltar = "conteudo/javascript/js - logica-de-programacao.html"
        var link_geral = "../../../index.html"
        var caminho_camada  = "../../../../"
        var icon = "../../../../assets"

      </script>
      
  </head>
  <body>
      <div class="header"></div>

    <main>
        <section>

            <div class="title"><h1 class="basico">Tratando erros (try, cath)</h1></div>
     
            <div class="container">
                


                    <div class="conteudo">
                        <p class="descricao">
                            Quando criamos um sistema, temos que tomar muito cuidado com os erro de programação, já que se houver algum, pode travar todo o sistema, então é muito perigoso e não pode haver falhas. Mas terá alguns momentos que teremos que fazer alguns testes ou não teremos certeza que um código funcionará, e para que o sistema não trave todo, caso aconteça algo errado, podemos fazer um tratamento de erros, que irá ajudar a amenizar o problema e evitar que o sistema caia. Para que possamos fazer o tratamento, iremos usar o try e catch, que tornará possível essa manipulação de código. O try e catch funcionam em conjunto, onde iremos colocar o código perigoso dentro do try, e se esse código apresentar erro, o try será bloqueado e o catch será executado, onde dentro do catch colocaremos o código seguro, o mais comum é colocar em uma variável e mandar para alguma base de dados, nunca, em hipotese alguma coloque o erro para ser exibido para o usuário. Para que possamos utilizar o try, teremos que declarar try{}, onde dentro dos parenteses iremos colocar o código que iremos testar, e logo em seguida, iremos colocar o catch(){}, onde dentro das chaves colocaremos o código que será executado caso dê problema no try e dentro dos parenteses iremos colocar um nome que carrega o erro, mas comum é colcocar o nome de error, que poderá ser usado para lançar o erro. 
                        </p>
                        <p class="descricao">

                        </p>
                    </div>
                    
                
                    <div class="codepen">
                        <p class="codepen" data-height="500" data-default-tab="js" data-slug-hash="xxXYOMj" data-user="kalilaziz" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;">
                            <span>See the Pen <a href="https://codepen.io/kalilaziz/pen/xxXYOMj">
                            063 - Tratando e lançando erros (try, catch, throw)</a> by KalilAziz (<a href="https://codepen.io/kalilaziz">@kalilaziz</a>)
                            on <a href="https://codepen.io">CodePen</a>.</span>
                          </p>
                          <script async src="https://cpwebassets.codepen.io/assets/embed/ei.js"></script>
                    </div>
        
                    
                
            </div>
    
            
    
        </section>
        <section>

            <div class="title"><h1 class="basico">Lançando erros (try, cath, throw)</h1></div>
     
            <div class="container">
                


                    <div class="conteudo">
                        <p class="descricao">
                            O tratamento de erros foi mostrado anteriormente como ser feito, mas para conseguirmos lançar esse erro, iremos ter que utilizar o throw. O throw normalmente será utilizado dentro de um if, onde colocamos como codição de execução algo que desviaria do que é esperado. para conseguirmos usar o throw, temos que declarar throw, posterormente temos que colocar new por ser um construtos e por fim, colocaremos o nome do erro juntamente com parenteses, sendo o mais comum colocar o nome de Error. Dentro dos parenteses colocaremos alguma mensagem descrevendo o erro que aconteceu entre aspas, por ser uma String. O restante será da mesma fora. Iremos declarar o try{} e catch(){}. Lembrando que usaremos o parâmetro dentro do argumento do catch para carregar o erro, onde que se acontecer algum erro no sistema, esse erro será enviado para o parâmetro, para que possamos enviar para alguma base de dados e fazer o tratamento posteriormente.  
                        </p>
                        <p class="descricao">
                            Iremos fazer o exemplo de uma função que tem a função de somar dois números, que serão passados pelos argumentos na chamada da função e recebido pelos parâmtros. Colocaremos um If, para que caso algum dos números passados não seja do tipo Number, ela irá executar o throw, lançando esse erro para o parâmetro do catch, denominado "error" automaticamente, caso os números passas estejam corretos, a função irá retornar a soma.
                        </p>
                        
                    </div>
                    
                
                    <div class="codepen">
                        <p class="codepen" data-height="800" data-default-tab="js" data-slug-hash="rNGXoJV" data-user="kalilaziz" style="height: 300px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;">
                            <span>See the Pen <a href="https://codepen.io/kalilaziz/pen/rNGXoJV">
                            063.2 - Tratando e lançando erros (try, catch, throw)</a> by KalilAziz (<a href="https://codepen.io/kalilaziz">@kalilaziz</a>)
                            on <a href="https://codepen.io">CodePen</a>.</span>
                          </p>
                          <script async src="https://cpwebassets.codepen.io/assets/embed/ei.js"></script>
                    </div>

                    <div class="image">
                        <img src="../../../image/erro tratamento.png" alt="">
                    </div>
        
                    
                
            </div>
    
            
    
        </section>
    </main>
    
    <script src="../../../../modules/active.js"></script>
    <script src="../../../../modules/header.js"></script>
    <script src="../../../Javascript/script.js"></script>
</body>
</html>